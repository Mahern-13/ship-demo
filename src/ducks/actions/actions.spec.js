import configureStore from "redux-mock-store";

import * as Actions from "./index.js";

const mockStore = configureStore();
const store = mockStore();

const stop = {
  id: 1,
  name: "test",
  address: "test123",
  completed: false,
  verified: true
};

describe("redux actions", () => {
  beforeEach(() => {
    store.clearActions();
  });

  it("dispatches deleteStop and the correct payload", () => {
    store.dispatch(Actions.deleteStop(1));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches createStop and the correct payload", () => {
    store.dispatch(Actions.createStop(stop, null, null));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches initCreateStop and the correct payload", () => {
    store.dispatch(Actions.initCreateStop("test", "test123"));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches updateStop and the correct payload", () => {
    store.dispatch(Actions.updateStop(stop, null, null));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches initUpdateStop and the correct payload", () => {
    store.dispatch(Actions.initUpdateStop(stop));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches setEditingStep and the correct payload", () => {
    store.dispatch(Actions.setEditingStep(1));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches setCompletionOfStop and the correct payload", () => {
    store.dispatch(Actions.setCompletionOfStop(stop));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches setCreatingStopState and the correct payload", () => {
    store.dispatch(Actions.setCreatingStopState(true));
    expect(store.getActions()).toMatchSnapshot();
  });

  it("dispatches setDidAddRoute and the correct payload", () => {
    store.dispatch(Actions.setDidAddRoute(true));
    expect(store.getActions()).toMatchSnapshot();
  });
});
