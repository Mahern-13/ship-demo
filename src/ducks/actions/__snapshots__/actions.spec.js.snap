// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux actions dispatches createStop and the correct payload 1`] = `
Array [
  Object {
    "alert": undefined,
    "error": undefined,
    "stop": undefined,
    "type": "CREATE_STOP",
  },
]
`;

exports[`redux actions dispatches deleteStop and the correct payload 1`] = `
Array [
  Object {
    "id": 1,
    "type": "DELETE_STOP",
  },
]
`;

exports[`redux actions dispatches initCreateStop and the correct payload 1`] = `
Array [
  Object {
    "address": "test123",
    "name": "test",
    "type": "INIT_CREATE_STOP",
  },
]
`;

exports[`redux actions dispatches initUpdateStop and the correct payload 1`] = `
Array [
  Object {
    "stop": Object {
      "address": "test123",
      "completed": false,
      "id": 1,
      "name": "test",
      "verified": true,
    },
    "type": "INIT_UPDATE_STOP",
  },
]
`;

exports[`redux actions dispatches setCompletionOfStop and the correct payload 1`] = `
Array [
  Object {
    "stop": Object {
      "address": "test123",
      "completed": false,
      "id": 1,
      "name": "test",
      "verified": true,
    },
    "type": "COMPLETION_OF_STOP",
  },
]
`;

exports[`redux actions dispatches setCreatingStopState and the correct payload 1`] = `
Array [
  Object {
    "bool": true,
    "type": "CREATING_STOP_STATE",
  },
]
`;

exports[`redux actions dispatches setDidAddRoute and the correct payload 1`] = `
Array [
  Object {
    "bool": true,
    "type": "SET_DID_ADD_ROUTE",
  },
]
`;

exports[`redux actions dispatches setEditingStep and the correct payload 1`] = `
Array [
  Object {
    "id": 1,
    "type": "EDITING_STEP",
  },
]
`;

exports[`redux actions dispatches updateStop and the correct payload 1`] = `
Array [
  Object {
    "alert": undefined,
    "error": undefined,
    "stop": undefined,
    "type": "UPDATE_STOP",
  },
]
`;
